free/src/test/scala/cats/free/YonedaTests.scala

```diff
@@ -0,0 +1,30 @@
+package cats
+package free
+
+import cats.tests.CatsSuite
+import cats.laws.discipline.{ArbitraryK, FunctorTests, SerializableTests}
+
+import org.scalacheck.Arbitrary
+import org.scalacheck.Prop.forAll
+
+class YonedaTests extends CatsSuite {
+  implicit def yonedaArbitraryK[F[_] : Functor](implicit F: ArbitraryK[F]): ArbitraryK[Yoneda[F, ?]] =
+    new ArbitraryK[Yoneda[F, ?]]{
+      def synthesize[A: Arbitrary]: Arbitrary[Yoneda[F, A]] =
+        Arbitrary(F.synthesize[A].arbitrary.map(Yoneda(_)))
+    }
+
+  implicit def yonedaArbitrary[F[_] : Functor : ArbitraryK, A : Arbitrary]: Arbitrary[Yoneda[F, A]] = yonedaArbitraryK[F].synthesize[A]
```

Comment:
I put it in the test to be the same as that of `FreeTests` - I think the reason we put them here is because `tests` does not depend on `free` at the moment, which leads us to doing strange things like placing `Arbitrary` instances inside the test. Perhaps we want `tests` to `dependOn(free)` ?